commons/src/main/java/org/apache/mesos/elasticsearch/common/zookeeper/ZookeeperCLIParameter.java

```diff
@@ -20,4 +20,12 @@ public String getZookeeperUrl() {
     public long getZookeeperTimeout() {
         return zookeeperTimeout;
     }
+
+    public static final String ZOOKEEPER_FRAMEWORK_URL = "--zookeeperFrameworkUrl";
+    @Parameter(names = {ZOOKEEPER_FRAMEWORK_URL}, required = false, description = "Zookeeper urls for the framework in the format zk://IP:PORT,IP:PORT,...)", validateWith = CLIValidators.NotEmptyString.class)
+    private String zookeeperFrameworkUrl = "zk://mesos.master:2181";
```

Comment:
I suggest defaulting this value to blank "", and then in the getZookeeperFrameworkURL, check if blank, and if so, use getZookeeperURL(). 
Because later in the code, you pass configuration.getFrameworkZKURL() to the executor. If the user hasn't set this, then the executor will always get mesos.master, rather than the getZookeeperURL.
And because you have set this as NOT REQUIRED, it is possible for the user not to set it.